cdk/lib/manifests.ts

```diff
@@ -0,0 +1,260 @@
+import { Deployment } from "kubernetes-types/apps/v1";
+import { Namespace, Service } from "kubernetes-types/core/v1";
+import * as yaml from "yaml";
+
+export type Manifests = {
+  server: ServiceDeployment;
+  worker: Deployment;
+  redis: ServiceDeployment;
+  namespace: Namespace;
+};
+
+export type ServiceDeployment = { service: Service; deployment: Deployment };
+
+type Env = "prod" | "dev";
+
+const LLMInABoxImage = "llm-in-a-box";
+const Namespace = "llm-in-a-box";
+
+/** Get the manifests as a javascript object */
+export const manifests = (environment: Env, numWorkers = 1): Manifests => {
+  const serverLabels = { app: "server" };
+  const serverPort = 8000;
+  const workerLabels = { app: "worker" };
+  const redisLabels = { app: "redis" };
+  const redisPort = 6379;
+
+  return {
+    namespace: {
+      apiVersion: "v1",
+      kind: "Namespace",
+      metadata: {
+        name: Namespace,
+      },
+    },
+    server: {
+      deployment: serverDeployment(serverLabels, serverPort, environment),
+      service: serverService(serverLabels, serverPort, environment),
+    },
+    worker: workerDeployment(workerLabels, environment, numWorkers),
+    redis: {
+      deployment: redisDeployment(redisLabels, redisPort),
+      service: redisService(redisLabels, redisPort),
+    },
+  };
+};
+
+/** Dump the manifests as a string */
```

Comment:
these manifests are the same as what was in kustomize, except the namespace is set explicitly to the value from the namespace variable.